### PR TITLE
feat: stub in /hcp/docs content

### PR DESCRIPTION
**Pull request by zchsh**
## 🔗 Relevant links

- [Preview link - /hcp/docs][/hcp/docs] 🔎
- [Asana task - Determine if HCP needs additional engineering work to migrate][task-explore] 🎟️
- [Asana task - Migrate HCP docs][task-migrate-hcp-docs] 🎟️

## 🗒️ What

Adds `/hcp/docs` routes.

## 🤷 Why

To explore potential rough edges in integrating HCP documentation into the HashiCorp Developer site.

## 🛠️ How

Adds `src/pages/hcp/docs`, similar to existing docs routes for other products.

## 🧪 Testing

- Visit [the /hcp/docs route][/hcp/docs], and ensure routes render as expected
    - Sidebar and sidecar contents should look and function as expected
    - The mobile menu should contain the right links and panels, and work as expected
    - Images should load
    - Search should work, and should search within HCP documentation only

## Other notes

- **Redirects have not been added, yet**. The content source repository is private, and we're still working on potential solutions to address this issue. Intent is to handle the redirects concerns outside of this PR. Relevant links:
    - 🎟️ [[redirects] load redirects from private repositories](https://app.asana.com/0/1202097197789424/1202532915796679/f)
    - We may also consider frontmatter-style redirects for docs files, like Learn does. Context in 🎟️ [Tutorials Redirects Property -> `old_filenames`](https://app.asana.com/0/1202097197789424/1201956804128029/f)
    - Note as well that existing dev-dot routes do not redirect as they do on dot-io sites. This is another reason why it felt reasonable to defer this work outside this PR. See 🎟️ [Get remotely sourced redirects working for dev-portal UI routes](https://app.asana.com/0/1201987349274776/1201662082096106/f)
    - As a fallback solution here, we could move the redirects.js file we need from the cloud.hashicorp.com repository into this hashicorp/dev-portal repository. This would come with author visibility concerns, but would at least "work" in the most basic sense of the word.
- **Only the `/docs` route has been stubbed in**. This is the only "MDX content" route on <https://cloud.hashicorp.com>. But, <https://cloud.hashicorp.com> has many other pages. The assumption at this time is that we don't plan to migrate those pages into the HashiCorp Developer site.
- **`Edit on GitHub` link is still shown**. This should likely be removed for HCP docs. However, there's currently no way to do this without changing the interface for `SidebarSidecarLayout` server utilities. Intent is to defer this tweak until after we've refactored that layout to extract docs-view specific logic.

[task-migrate-hcp-docs]: https://app.asana.com/0/1201987349274776/1201650734121659/f
[task-explore]: https://app.asana.com/0/1202097197789424/1202371394012663/f
[/hcp/docs]: https://dev-portal-git-zsstub-hcp-docs-hashicorp.vercel.app/hcp/docs

### CODE REVIEW

**Comment by zchsh** [comment]
Update here: images are still not working, I think due to some subtleties with `next/image`? I need to investigate further.
- Images seem to load fine directly from the content API, for example `hcp-arch-diagram.png` can be loaded [like this](https://content.hashicorp.com/api/assets?product=cloud.hashicorp.com&version=refs/heads/main&asset=public/img/docs/hcp-arch-diagram.png) or [like this with Imgix parameters](https://content.hashicorp.com/api/assets?product=cloud.hashicorp.com&version=refs/heads/main&asset=public/img/docs/hcp-arch-diagram.png?width=994&height=645&w=2048&q=75)
- Images do not seem to load on this preview, eg [this next/image path does not load](https://dev-portal-git-zsstub-hcp-docs-hashicorp.vercel.app/_next/image?url=https%3A%2F%2Fcontent.hashicorp.com%2Fapi%2Fassets%3Fproduct%3Dcloud.hashicorp.com%26version%3Drefs%252Fheads%252Fmain%26asset%3D%252Fpublic%252Fimg%252Fdocs%252Fhcp-arch-diagram.png%26width%3D994%26height%3D645&w=2048&q=75)

File: src/views/product-root-docs-path-landing/index.tsx [hunk 1]
```diff
@@ -20,7 +20,20 @@ const ProductRootDocsPathLanding = ({
...
 
   let mdxSlot: ReactElement
-  if (mdxSource) {
+  /**
+   * TODO: cloud.hashicorp.com/docs redirects to cloud.hashicorp.com/docs/hcp,
+   * and the content for the former URL is empty (just a heading).
+   * This is a workaround to prevent the empty content from being rendered.
+   * https://github.com/hashicorp/cloud.hashicorp.com/blob/main/content/docs/index.mdx
+   *
+   * Related redirects-in-dev-dot issues:
```
Comment (zchsh) [inline src/views/product-root-docs-path-landing/index.tsx:29]: Have noted in PR description as well, but for now, redirects authored in cloud.hashicorp.com are not being applied. Intent is to defer this work outside this PR.

File: src/views/product-root-docs-path-landing/server.ts [hunk 2]
```diff
@@ -94,6 +95,7 @@ const generateGetStaticProps = ({
     const { getStaticProps: generatedGetStaticProps } =
       _getStaticGenerationFunctions({
         product,
+        productSlugForLoader,
```
Comment (zchsh) [inline src/views/product-root-docs-path-landing/server.ts:98]: For HCP docs, the HashiCorp Developer product slug `hcp` does not match the `cloud.hashicorp.com` value we need to send to the content API.

**Comment by zchsh** [comment]
@BRKalow @ashleemboyer This PR stubs in `/hcp/docs` content. Aim here is mainly to explore rough edges, and get something that works-for-now in place (definitely not targeting this as "HCP stuff is done!" - very far from it!). 

For additional context, this did yield some edge-y cases in the content API that required a couple tweaks (eg to get images to load). I worked together with Kevin (and Bryce) to patch those up in:
- https://github.com/hashicorp/mktg-content-workflows/pull/191
- https://github.com/hashicorp/mktg-terraform/pull/15
- https://github.com/hashicorp/mktg-terraform/pull/16
- https://github.com/hashicorp/mktg-content-workflows/pull/198

Let me know what ya'll think 🙇‍♂️

File: src/lib/get-deployed-url.ts [hunk 3]
```diff
@@ -11,7 +11,7 @@ export default function getDeployedUrl() {
   }
 
   if (process.env.HASHI_ENV === 'development') {
-    return ''
+    return 'http://localhost:3000'
```
[the remark this answers is not in the document]
Comment (BRKalow) [reply src/lib/get-deployed-url.ts:14]: edge case note: if someone has more than one next app running locally, the port might not be 3000.

[the remark this answers is not in the document]
Comment (zchsh) [reply src/lib/get-deployed-url.ts:14]: @BRKalow ah very true... i suppose setting an env var for `PORT` in our `npm run start` command could in one way help address this (static port), but in another way mess up the automatic "use a different port" logic...

nonetheless, maybe worth floating! How would we feel about doing that, eg would something like this work?

In `package.json`:

```json
{
  "scripts": {
    "start": "PORT=3000 next"
  }
}
```

Here in `get-deployed-url`:

```js
if (process.env.HASHI_ENV === 'development') {
    return `http://localhost:${process.env.PORT}`
}
```

I guess we'd also need to pass `PORT` through in `next.config.js`?

```
env: {
  /* ... */
  PORT: process.env.PORT,
  /* ... */
},
```

The only other solution I can think of is to not error in development when `props.image` in `react-head` is not absolute... but then I think we lose the intent behind that change 🤔 maybe okay given the above "fixed port" compromise? 🤷

**Review by ashleemboyer — APPROVED** [review]
LGTM! 🛹 ✨

File: src/lib/get-deployed-url.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (zchsh) [reply src/lib/get-deployed-url.ts:14]: Alternate solution for this: https://github.com/hashicorp/dev-portal/pull/676

**Comment by zchsh** [comment]
Note: had to fix some conflicts with `main` due to `spaces → tabs` change upstream. Also moved the local dev issue to a separate PR, #676, which has sinced merged. This PR's contents are unchanged otherwise.

**Pull request merged**